### ISSUE TITLE
Out of memory on table creation statement

### ISSUE THREAD

**Issue by ngould** [duckdb/duckdb#5315]
### What happens?

DuckDB produces the following out of memory error when I attempt to create a table derived from an event log with ~164M rows, as follows:

Error: `Error: Out of Memory Error: could not allocate block of 262144 bytes (27487965184/27487790694 used)`

SQL statement:
```
CREATE TABLE IF NOT EXISTS first_page_view as (
  SELECT
    user_id,
    page_id,
    min(viewed_at) as first_view,
  FROM page_views
  GROUP BY user_id, page_id
);
```

The same query works fine if executed on its own, rather than in the context of a `CREATE TABLE` statement.

### To Reproduce

You can grab the exact dataset I'm using here if that's helpful. (It's a 6.3 GB parquet file.)

https://duckdb-example-data.s3.amazonaws.com/example.parquet

### OS:

OSX (x86)

### DuckDB Version:

v0.5.2-dev2236

### DuckDB Client:

CLI

### Full Name:

Nathan Gould

### Affiliation:

Endeavor Labs

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by ngould** [comment]
FYI this is on a machine with 16 GB RAM, using default duckdb memory configs.

**Comment by Mytherin** [comment]
Is this with a persistent database or with an in-memory database?

**Comment by ngould** [comment]
Discussed in Discord. 

https://discord.com/channels/909674491309850675/921073327009853451/1041091126435794954

TL;DR: This is expected behavior, as DuckDB's hash aggregate requires the entire result set to fit in memory. As a partial workaround, removing the memory limit (`SET memory_limit: -1;`) may help, especially on MacOS where support for compressed memory may result in unnecessarily restrictive DuckDB default memory limits.

Possible improvements suggested by @Mytherin:
1. Increasing default DuckDB memory limit on MacOS.
2. Add a more instructive error message

**Comment by omo** [comment]
> TL;DR: This is expected behavior, 

@ngould Maybe you can close this. Thanks for your comment, people will later find the solution when they encountered the same problem.
